drivers/platform/aducm3029/spi.c

```diff
@@ -0,0 +1,228 @@
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/************************* Include Files **************************************/
+/******************************************************************************/
+
+#include "spi_extra.h"
+#include "spi.h"
+#include "error.h"
+#include <stdlib.h>
+
+/******************************************************************************/
+/*****************************  Variables   **********************************/
+/******************************************************************************/
+
+static uint8_t	active_spi = 0; //Stores the active SPIs
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+
+/**
+ * @brief Allocates the memory needed for the descriptor
+ * @return A address to the allocated memory, NULL if the allocation fails.
+ */
+static struct spi_desc *alloc_data_mem(void)
+{
+	struct spi_desc		*desc;
+	struct adicup_spi_desc	*adicup_desc;
+
+	desc = calloc(1, sizeof(struct spi_desc));
+	if (!desc)
+		return NULL;
+
+	adicup_desc = calloc(1, sizeof(struct adicup_spi_desc));
+	if (!adicup_desc) {
+		free(desc);
+		return NULL;
+	}
+	desc->extra = adicup_desc;
+
+	adicup_desc->buffer = calloc(1, ADI_SPI_MEMORY_SIZE);
+	if (NULL == adicup_desc->buffer) {
+		free(adicup_desc);
+		free(desc);
+		return NULL;
+	}
+
+	return desc;
+}
+
+/**
+ * @brief Deallocates the memory needed for the descriptor
+ * @param desc - Descriptor to be deallocated
+ */
+static void free_desc_mem(struct spi_desc *desc)
+{
+	free(((struct adicup_spi_desc*)(desc->extra))->buffer);
```

Comment:
I would check here for each pointer if deallocation is required.

Maybe use `adicup_spi_desc` type pointer to avoid casting.